pallene/coder.lua

```diff
@@ -519,16 +540,35 @@ function Coder:lua_entry_point_definition(f_id)
 
     local arg_vars  = {}
     local arg_decls = {}
+    for i, captured_var in ipairs(captured_vars) do
+        local name = self:c_upval(i)
+        table.insert(arg_vars, name)
+        table.insert(arg_decls, C.declaration(ctype(captured_var.decl.typ), name)..";")
+    end
     for i, typ in ipairs(arg_types) do
-        local name = "x"..i
-        arg_vars[i] = name
-        arg_decls[i] = C.declaration(ctype(typ), name)..";"
+        local name = self:c_var(i)
+        table.insert(arg_vars, name)
+        table.insert(arg_decls, C.declaration(ctype(typ), name)..";")
     end
 
+
     local init_args = {}
+    local n_initialized_args = 0
```

Comment:
Could we get rid of the `n_initialized_args` if we used  `table.insert(arg_vars`?

Reply:
I'm not sure about that. Currently `n_initialized_args` is being used in two places:
In the first loop:
```lua
local src  = string.format("&func->upvalue[%s]", C.integer(n_initialized_args - 1))
```
and in the second loop:
```lua
local dst = arg_vars[n_initialized_args]
```

I can't think of a way to get rid of it :sweat_smile: .

Reply:
Good question.

For the `func->upvalue[%s]`, I think it would be clearer if we used the upvalue id instead of `n_initialized - 1`.

For the `arg_vars`, what do you think about splitting it into two arrays? One list for the captured variables and one list for the function arguments?

Reply:
Ah, good idea. I forgot `captured_Vars` is a list now